app/views/posts/_post.html.erb

```diff
@@ -0,0 +1,37 @@
+<% 
+  show_comments = !current_page?(user_path(post.author.id)) 
+  show_truncate = !current_page?(user_post_path(post.author.id, post.id)) 
+  if show_truncate
+    post_text = "#{post.text[0..150]}..."
+  else
+    post_text = post.text
+  end
+%>
+
+<div class="box-no-border">
+  <div class="column box">
+    <% if show_truncate %>
+      <h3><%= link_to(user_post_path(post.author.id, post.id)) { post.title } %></h3>
+    <% else %>
+      <div class="row-justify">
+        <h3><%= post.title %> by <%= link_to(user_path(post.author.id)) { post.author.name } %> </h3>
+        <h4><%= render partial: 'posts/count', object: post %></h4>
+      </div>
+    <% end %>
+    <p><%= post_text %></p>
+    <% if show_truncate %>
+      <div class="row-right">
+        <h4><%= render partial: 'posts/count', object: post %></h4>
+      </div>
+    <% end %>
+  </div>
+  <% if show_comments && post.comments_count > 0 %>
+    <div class="column box">
+      <% if show_truncate %>
+        <%= render post.recent_comments %>
```

Comment:
- [ ] It would be better if this bug is fixed so the see all post button works as expected
![Screenshot from 2022-08-18 16-16-20](https://user-images.githubusercontent.com/46388113/185445540-86818d41-0fef-47a6-90e2-1a1395d6b3cc.png)

Reply:
@addod19 Please could you make this clearer because i don't see this error from my end.

Reply:
This error can as a result of a user clicking on the “See all post” link. I’m sure you want the user to see all the post once you click on the link but it returned the error you see in the screenshot. I hope this is clearer

Reply:
Oh Okay.

It is.
Thank you